refinery/tool_manager/tests.py

```diff
@@ -2108,6 +2108,32 @@ def test_attach_outputs_dataset_makes_proper_node_inheritance_chain(self):
 
         self._assert_analysis_node_connection_outputs_validity()
 
+    def test_galaxy_renamedatasetaction_handling(self):
+        new_dataset_name = "COFFEE"
+        workflow_step = 1
+        workflow_dict = galaxy_workflow_dict
+        workflow_dict["steps"][str(workflow_step)]["post_job_actions"] = {
+            "RenameDatasetActionRefinery test tool LIST - N on data 4": {
+                "action_arguments": {
+                    "newname": new_dataset_name
+                },
+                "action_type": "RenameDatasetAction",
+                "output_name": "Refinery test tool LIST - N on data 4"
+            }
+        }
+        self.get_workflow_dict_mock.return_value = workflow_dict
+        self.galaxy_datasets_list_mock.start()
+
+        self.create_tool(ToolDefinition.WORKFLOW)
+        galaxy_datasets = self.tool._get_galaxy_history_dataset_list()
+        edited_galaxy_dataset = next(
+            galaxy_dataset for galaxy_dataset in galaxy_datasets if
+            self.tool._get_workflow_step(galaxy_dataset) == workflow_step
+        )
```

Comment:
I don't know that it's an interesting test, but what would you expect with just the comprehension? If you're not not really traversing the data structure, next is a little odd.